### PR TITLE
app/vminsert: fixes readonly check

### PR DESCRIPTION
**Pull request by f41gh7**
previously vminsert doesn't check readOnly state for vmstorage, since check was never performed for nil buffer In this case every 30 second storage node loss readonly state and received some data. It caused re-routing and possible slow down for ingestion https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4870

### CODE REVIEW

**Review by zekker6 — APPROVED** [review]

File: app/vminsert/netstorage/netstorage.go [hunk 1]
```diff
@@ -319,11 +319,15 @@ var cannotCloseStorageNodeConnLogger = logger.WithThrottler("cannotCloseStorageN
 
 var cannotSendBufsLogger = logger.WithThrottler("cannotSendBufRows", 5*time.Second)
 
-func sendToConn(bc *handshake.BufferedConn, buf []byte) error {
+func sendToConnBuf(bc *handshake.BufferedConn, buf []byte) error {
```
Comment (hagen1778) [inline app/vminsert/netstorage/netstorage.go:322]: maybe we don't need this func at all? 

the bc is checked for being nil right before the call of this function on L:282. It has no other usages.

Comment (f41gh7) [reply app/vminsert/netstorage/netstorage.go:322]: function `sendToConnBuf` checks len of buf. If it's zero, fast path trigger with early return.

function `sendToConn` do not perform this check anymore and may send an empty buf. 

 It's needed for vmstorage read-only check . function `checkReadOnlyMode` uses `sendToConn`  with zero len buf for reading state response from vmstorage.

Comment (hagen1778) [reply app/vminsert/netstorage/netstorage.go:322]: Yes, but if you remove only `!=nil` check from `sendToConn` - it should do the thing, right?

Comment (f41gh7) [reply app/vminsert/netstorage/netstorage.go:322]: Yep, but it may have ingestion performance impact.

Comment (hagen1778) [reply app/vminsert/netstorage/netstorage.go:322]: why?

Comment (f41gh7) [reply app/vminsert/netstorage/netstorage.go]: I've double checked vminsert code and the same check already performed by the code  at `(sn *storageNode) run()` method:
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/app/vminsert/netstorage/netstorage.go#L180

 Simplified fix.

**Review by hagen1778 — APPROVED** [review]
LGTM

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

**Comment by valyala** [comment]
@f41gh7 , thanks for the fix!

**Pull request merged**